engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/EnsoHashMap.java

```diff
@@ -66,7 +66,7 @@ EnsoHashMapBuilder getMapBuilder(
     }
   }
 
-  Object getCachedVectorRepresentation() {
+  public Object getCachedVectorRepresentation() {
```

Comment:
Maybe there was a reason why the method wasn't public! It does not have any reasonable performance. I see `NewArray` allocation coming from this method in the graphs. The usage of `ConditionProfile.getUncached()` probably also doesn't help. The size of the graphs - they are huge!

Why do we want to obtain `Warning[]` as array when it is enough to have them as a snapshot inside of the `EnsoHashMap` anyway?

Reply:
Please try to get the API of `hash` (and also `vector`) packages where it used to be. The objects shall be opaque. Operations on them should be accessible thru nodes.

Reply:
All public methods of `EnsoHashMap` and `EnsoHashMapBuilder` refactored to package-protected in these commits:
- 39f7262a7ed6772d542d043be3d9ec63f7ec0353
- 2d7a8b3c571e6545d6ed8cdb065179f897019d70
- 0ecab16a50462fecb3ae1cfae0fcaca0c9f53ac3